Frontend/src/test/pages/dashboard/components/Customer/AddressCard.test.jsx

```
@@ -0,0 +1,103 @@
+vi.mock(
+  '../../../../../pages/dashboard/components/Customer/EditAddressDialog',
+  () => ({
+    EditAddressDialog: ({ address, open, onOpenChange, onSave }) => (
+      <div data-testid="edit-dialog" data-open={open ? 'true' : 'false'}>
+        <button
+          data-testid="save-btn"
+          onClick={() => {
+            onSave({ ...address, street: 'Edited Street' })
+            onOpenChange(false)
+          }}
+        >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)